pallets/snarcos/src/lib.rs

```diff
@@ -303,8 +303,14 @@ pub mod pallet {
                     )
                 })?;
 
-            let valid_proof = S::verify(&verification_key, &public_input, &proof)
-                .map_err(|err| (Error::<T>::VerificationFailed(err), None))?;
+            let parent_hash = <frame_system::Pallet<T>>::parent_hash();
+            let valid_proof = S::verify(
+                &verification_key,
+                &public_input,
+                &proof,
+                parent_hash.as_ref(),
```

Comment:
It's not _really_ random though as it can be influenced to some extent. Probably _good enough_ though.

Reply:
usually proof generation lasts longer than block-time, so yeah, good enough :P